File: CS/09_Forms/SetExportValueForCheckbox.cs
using System;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Widget;

namespace SetExportValueForCheckbox
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Input and output file paths
            string input = @"..\..\..\..\..\..\Data\SetExportValueForCheckbox.pdf";
            string result = "SetExportValueForCheckbox_result.pdf";

            // Create a new instance of PdfDocument
            PdfDocument pdf = new PdfDocument();

            // Load the PDF document from the specified file path
            pdf.LoadFromFile(input);

            // Get the form widget from the loaded PDF document
            PdfFormWidget formWidget = pdf.Form as PdfFormWidget;

            // Initialize a counter variable
            int count = 1;

            // Traverse all fields in the FieldsWidget
            foreach (PdfFieldWidget field in formWidget.FieldsWidget)
            {
                // Check if the field is a checkbox
                if (field is PdfCheckBoxWidgetFieldWidget)
                {
                    // Cast the field to PdfCheckBoxWidgetFieldWidget
                    PdfCheckBoxWidgetFieldWidget checkbox = field as PdfCheckBoxWidgetFieldWidget;

                    // Set the export value for the checkbox
                    checkbox.SetExportValue("True" + (count++));
                }
            }

            // Save the modified PDF document to the specified file in PDF format
            pdf.SaveToFile(result, FileFormat.PDF);

            //Show the result file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/09_Forms/SetExportValueForCheckbox.vb
Imports Spire.Pdf
Imports Spire.Pdf.Widget

Namespace SetExportValueForCheckbox
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Specify the input file path
            Dim input As String = "..\..\..\..\..\..\Data\SetExportValueForCheckbox.pdf"

            ' Create a new instance of PdfDocument
            Dim pdf As New PdfDocument()

            ' Load the PDF document from the specified input file
            pdf.LoadFromFile(input)

            ' Access the form widget of the PDF document
            Dim formWidget As PdfFormWidget = TryCast(pdf.Form, PdfFormWidget)

            ' Initialize a count variable to track export values
            Dim count As Integer = 1

            ' Iterate through each field in the form
            For Each field As PdfFieldWidget In formWidget.FieldsWidget

                ' Check if the field is a checkbox widget field
                If TypeOf field Is PdfCheckBoxWidgetFieldWidget Then
                    Dim checkbox As PdfCheckBoxWidgetFieldWidget = TryCast(field, PdfCheckBoxWidgetFieldWidget)

                    ' Set the export value for the checkbox field
                    checkbox.SetExportValue("True" & (count))

                    ' Increment the count variable
                    count += 1
                End If
            Next field

            ' Specify the output file path
            Dim result As String = "SetExportValueForCheckbox_result.pdf"

            ' Save the modified PDF document to the output file using PDF format
            pdf.SaveToFile(result, FileFormat.PDF)

            ' Close the document
            pdf.Close()

            ' Launch the result file
            PDFDocumentViewer(result)
        End Sub
        Private Sub PDFDocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class
End Namespace
